src/monitor/monitor.go

```diff
@@ -58,11 +59,12 @@ type Monitor struct {
 }
 
 // New creates monitor service
-func New(log logrus.FieldLogger, cfg Config, addrManager AddrManager, dpstget DepositStatusGetter, sag ScanAddressGetter) *Monitor {
+func New(log logrus.FieldLogger, cfg Config, addrManager, ethAddrManager AddrManager, dpstget DepositStatusGetter, sag ScanAddressGetter) *Monitor {
```

Comment:
Later want to wrap the AddrManager so that we don't add new argument to this for every coin.  For now this is ok here

Reply:
Or we can remove these from the arguments list and require the caller to assign the AddrManagers after creating

Reply:
yes, adding new argument is not good design

Reply:
done, i add a AddrManager